mev-relay-rs/src/relay.rs

```diff
@@ -411,51 +405,53 @@ impl Relay {
         Ok(())
     }
 
+    // NOTE: we assume that `auction_request` and `signed_submission` have been validated.
     fn insert_bid_if_greater(
         &self,
         auction_request: AuctionRequest,
-        execution_payload: ExecutionPayload,
+        signed_submission: &SignedBidSubmission,
         value: U256,
-        builder_public_key: BlsPublicKey,
     ) -> Result<(), Error> {
+        let builder_public_key = auction_request.public_key.clone();
```

Comment:
the `public_key` here refers to the proposer for this auction

and why we passed in the `builder_public_key` separately

Reply:
ah, I think you just meant to get this from the `signed_submission` instead, in which case we can keep the removal from the fn sig like you have